garden_ai/app/hpc_notebook.py

```diff
@@ -13,9 +20,62 @@ def hpc_notebook():
 
 
 @hpc_notebook_app.command()
-def start():
+def start(
+    working_directory: str = typer.Option(
+        ..., help="Working directory for the operation."
+    ),
+    notebooks_dir: str = typer.Option(..., help="Directory to bind for notebooks."),
+    container_image: str = "hpc-notebook.sif",
+    definition_file: str = "Singularity.def",
+):
     """Open a notebook file in HPC."""
-    print("🚧🌱🚧 Under Construction 🚧🌱🚧")
+
+    tmp_dir = os.path.abspath(os.path.join(working_directory, "tmp"))
+    notebooks_dir = os.path.abspath(notebooks_dir)
+
+    # Step 1: Create temporary directory if it doesn't exist
+    os.makedirs(tmp_dir, exist_ok=True)
+
+    # Step 2: Set environment variables
+    os.environ["SINGULARITY_TMPDIR"] = tmp_dir
+    os.environ["APPTAINER_TMPDIR"] = tmp_dir
+    os.environ["JUPYTER_RUNTIME_DIR"] = tmp_dir
+    os.environ["JUPYTER_DATA_DIR"] = tmp_dir
+    os.environ["JUPYTER_CONFIG_DIR"] = tmp_dir
+
+    try:
+        # Ensure the definition file exists
+        if not os.path.isfile(definition_file):
+            logger.error(f"Definition file {definition_file} not found.")
+            typer.echo(f"🚧🌱🚧 Definition file {definition_file} not found 🚧🌱🚧")
+            raise typer.Exit(code=1)
+
+        # Step 3: Build the Apptainer container
+        build_command = ["apptainer", "build", container_image, definition_file]
+        subprocess.run(build_command, check=True)
+        logger.info("Apptainer container built successfully.")
+
+        # Step 4: Run the Apptainer container and start Jupyter Notebook
+        run_command = [
+            "apptainer",
+            "run",
+            "--bind",
+            f"{notebooks_dir}:/notebooks",
+            container_image,
+            "jupyter",
+            "notebook",
+            "--no-browser",
+            "--ip=0.0.0.0",
+            "--allow-root",
+        ]
+        subprocess.run(run_command, check=True)
+        logger.info("Jupyter Notebook started successfully in the Apptainer container.")
+    except subprocess.CalledProcessError as e:
+        logger.error(f"Failed to start Jupyter Notebook: {e}")
+        typer.echo("🚧🌱🚧 Failed to start Jupyter Notebook 🚧🌱🚧")
+    except Exception as e:
+        logger.error(f"An error occurred: {e}")
+        typer.echo("🚧🌱🚧 An unexpected error occurred 🚧🌱🚧")
```

Comment:
How should the user shut down their Singularity container when they're done? It's ok to not build in really robust cleanup handling right now, but ideally we could handle the happy path of the user hitting ctrl-c. Or at least we should print instructions to the user for how they can manually end their session.